salt/states/file.py

```diff
@@ -6905,6 +6911,16 @@ def _patch(patch_file, options=None, dry_run=False):
             reverse_pass = _patch(patch_rejects, ["-R", "-f"], dry_run=True)
             already_applied = reverse_pass["retcode"] == 0
 
+            # Check if the patch command threw an error upon execution
+            # and return the error here. According to gnu on patch-messages
+            # patch exits with a status of 0 if successful
+            # patch exits with a status of 1 if some hunks cannot be applied
+            # patch exits with a status of 2 if something else went wrong
+            # www.gnu.org/software/diffutils/manual/html_node/patch-Messages.html
+            if pre_check["retcode"] == 2 and pre_check["stderr"]:
```

Comment:
When I made exit code 1 return a failure status in the salt return a failure tests in `tests/integration/states/test_file.py::PatchTest` started failing. Apparently partially applied patches have always returned as a successful job. I don't think we can change this without some kind of deprecation period. I don't think it's worth doing that since (to my knowledge) no one has complained about the way it's working now.

Reply:
Oh. That works for me then.